api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -43,8 +47,18 @@ public CohortMembership(Patient patient) {
 		this(patient, new Date());
 	}
 	
-	public boolean isMemberActive() {
-		return this.getStartDate() != null && this.getStartDate().before(new Date()) && this.getEndDate() == null;
+	public boolean isMembershipActive() {
+		return this.getStartDate() != null &&
+				this.getStartDate().before(new Date()) &&
+				this.getEndDate() == null &&
+				!this.getVoided();
+	}
+	
+	public boolean isMembershipActive(Date asOfDate) {
+		return this.getStartDate() != null &&
+				!this.getStartDate().after(asOfDate) &&
```

Comment:
!voided && the asOfDate.after(startDate) && (endDate == null || asOfDate.before(endDate)